### ISSUE TITLE
Feature Request: On reload file triggers close and open notifications

### ISSUE THREAD

**Issue by Ekopalypse** [notepad-plus-plus/notepad-plus-plus#6038]
### Description of the Issue
Npp sends only a BUFFERACTIVATED notification when either using **reload from disc** from file menu or **reload** from tab context menu. 
As there is no additional info in the notification, other than the bufferid,
a plugin can not know if the buffer has been activated by switching a tab or 
because of a file reload action as mentioned before.
In order to provide this knowledge to plugins it is needed to get the notifications which are
sent by doing normal file open and close actions.

Maybe there is another way, which I missed, on how to get the info that the file has been reloaded?

### Steps to Reproduce the Issue
1. load a file
2. make some changes
3. reload the file
4. observer notifications sent

### Expected Behavior
receiving notifications in the order
FILEBEFORECLOSE
FILECLOSED
FILEBEFORELOAD
FILEBEFOREOPEN
FILEOPENED
BUFFERACTIVATED

### Actual Behavior
only BUFFERACTIVATED is sent

### Debug Information
Notepad++ v7.7.1   (64-bit)
Build time : Jun 16 2019 - 21:24:47
Path : ...\notepad++.exe
Admin mode : OFF
Local Conf mode : OFF
OS : Windows 7 (64-bit)
Plugins : test1.dll

**Comment by dinkumoil** [comment]
@Ekopalypse 

It is not enough to send these messages you mentioned above to plugins. Additionally all the steps of closing, opening and loading a file should be processed. Currently this is not the case.

For example:

1. Open a file whose encoding you know for sure is ANSI, preferably one that contains characters with character codes greater than 127 (extended ASCII range).
2. Click on `(menu) Encoding -> Encode in UTF-8`. The encoding is changed to UTF-8 and the characters with a code greater than 127 are displayed wrong.
3. Click on `(menu) File -> Reload from Disk`. The file is supposed to be reloaded but the characters with a code greater than 127 are still displayed wrong.

This is not a real file reload.

**Comment by Ekopalypse** [comment]
@dinkumoil 
You are right, actually I thought "closing" and "opening" will be done under the hood when reloading but this seems not to be the case.
The more I think about it, closing seems not to be necessary, as there is nothing to close, on the other side, it might be necessary to clean up certain stuff but I'm unsure about that. But running the opening procedure like done with a normal open file action is needed.

**Comment by dinkumoil** [comment]
@Ekopalypse 

> closing seems not to be necessary

No, it **is** necessary. I had to include in all my plugins a buffer management to remember for which buffer I've already switched character encoding (_AutoCodapge_ plugin), EOL format (_AutoEolFormat_ plugin) and line number format (_CustomLineNumbers_ plugin). I use the _FILECLOSED_ event to remove a certain buffer from this management system. Without that event this wouldn't be possible.

Maybe other plugin authors have similar needs.

**Comment by alankilborn** [comment]
Bump.  I noticed the problem described in this issue when I was scripting something yesterday.  I want to do a "time-consuming" (on larger files) action when a reloaded file occurs.  Doing it upon every "buffer activated" would be too annoying to the user.
